imap/index.c

```diff
@@ -5226,18 +5227,25 @@ static int ping(struct backend *s,
     /* Read response(s) from backend until final response or error */
     do {
         resp_body.flags = BODY_DISCARD;
-        if (http_read_response(s, METH_HEAD, &statuscode,
-                               &resp_hdrs, &resp_body, &errstr)) {
+        r = http_read_response(s, METH_HEAD, &statuscode,
+                               &resp_hdrs, &resp_body, &errstr);
+        if (r) {
             break;
         }
     } while (statuscode < 200);
 
-    syslog(LOG_DEBUG, "extractor_ping: HEAD %s: got status %u",
-           ctx->path, statuscode);
+    if (r) {
+        syslog(LOG_DEBUG, "extractor_ping: HEAD %s: failed (%d): %s",
+               ctx->path, r, errstr);
```

Comment:
instead of `%d` and `r`, I'd suggest `%s` and `error_message(r)` is more useful to have in the log

Reply:
I agree with ellie.  Otherwise, this should be good

Reply:
Fixed, and thanks. The more you know!